### PR TITLE
PMM-10121 split CI jobs

### PR DESCRIPTION
**Pull request by idoqo**
PMM-10121

This splits the CI jobs into `CI` for general changes, `pmm-admin CI` for admin-related changes, and `pmm-agent CI` for agent-related changes.

It also introduces a new Action ([`merge-gatekeeper`](https://github.com/percona-platform/merge-gatekeeper/)) that gets the status of all executed jobs. The Slack notification is then based on the result of merge-gatekeeper.

I checked with three commits:
- https://github.com/percona/pmm/pull/922/commits/7f1431e6140b441f3c1c24129731c789c1c176d0 doesn't change admin or agent, so both pmm-admin CI and pmm-agent CI are ignored. `merge-gatekeeper` is successful and CI sends a [Slack success message](https://perconacorp.slack.com/archives/G4VC14WTF/p1654672416214449).
- https://github.com/percona/pmm/pull/922/commits/f55c6b2177e366e5bdfa6c66481542148b8c3dda changes admin, and the checks are all successful. `merge-gatekeeper` is successful and CI sends [Slack success message](https://perconacorp.slack.com/archives/G4VC14WTF/p1654673408050859).
- https://github.com/percona/pmm/pull/922/commits/b638f3230a2cf86e583527e420af8368713f1e7a changes admin and agent, but agent fails (due to failing MariaDB tests). `merge-gatekeeper` failed and CI sends [Slack failure message](https://perconacorp.slack.com/archives/G4VC14WTF/p1654674099598519).

Build: SUBMODULES-0

### CODE REVIEW

File: agent/README.md [hunk 1]
```diff
@@ -8,20 +8,18 @@ pmm-agent for PMM 2.x.
...
 * mongodb_exporter
 * proxysql_exporter
 
-Run `make init` to install dependencies.
-
 #### To run pmm-agent
-Run [PMM-server](https://github.com/percona/pmm) docker container or [pmm-managed](https://github.com/percona/pmm-managed).  
-Run `make setup-dev` to configure pmm-agent
-Run `make run` to run pmm-agent
+- Run [pmm-server docker container](https://hub.docker.com/r/percona/pmm-server) or [pmm-managed](https://github.com/percona/pmm-managed).
```
Comment (BupycHuk) [inline agent/README.md:20]: 👍🏼

**Review by BupycHuk — APPROVED** [review]

File: .github/workflows/ci-admin.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    paths:
+      - "admin/**"
+    branches:
+      - main
+      - release-*
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+  pull_request:
+    paths:
+      - "admin/**"
```
Comment (artemgavrilov) [inline .github/workflows/ci-admin.yml:14]: We should  run CI for admin if API changed or something in the root of the project

File: .github/workflows/ci-admin.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+name: pmm-admin CI
+
+on:
+  push:
+    paths:
+      - "admin/**"
```
Comment (artemgavrilov) [inline .github/workflows/ci-admin.yml:6]: For the `push` I thing better always to test everything

File: .github/workflows/ci-agent.yml [hunk 4]
```diff
@@ -0,0 +1,145 @@
+name: pmm-agent CI
+
+on:
+  push:
+    paths:
+      - "agent/**"
```
Comment (artemgavrilov) [inline .github/workflows/ci-agent.yml:6]: Same as for agent

File: .github/workflows/ci-agent.yml [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    branches:
+      - main
+      - release-*
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+  pull_request:
+    paths:
+      - "agent/**"
```
Comment (artemgavrilov) [inline .github/workflows/ci-agent.yml:14]: Same as for agent

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -100,203 +100,16 @@ jobs:
...
-      - name: Check out code into the working directory under the GOPATH
-        uses: percona-platform/checkout@v3
-        with:
-          fetch-depth: 1
-          path: ${{ env.WORKING_DIR }}
-
-      - name: Set up Go release
-        uses: percona-platform/setup-go@v2
+      - name: Run Merge Gatekeeper
+        uses: percona-platform/merge-gatekeeper@main
```
Comment (artemgavrilov) [inline .github/workflows/ci.yml:107]: Let's use exact version instead of `main`

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -100,203 +100,16 @@ jobs:
...
-      GOPATH: ${{ github.workspace }}
-      GOBIN: ${{ github.workspace }}/bin
-      WORKING_DIR: ${{ github.workspace }}/src/github.com/percona/pmm
-
-    defaults:
-      run:
-        working-directory: ${{ env.WORKING_DIR }}/admin
-
+  merge-gatekeeper:
+    runs-on: ubuntu-latest
```
Comment (artemgavrilov) [inline .github/workflows/ci.yml:104]: Let's use same version as in other jobs

**Review by artemgavrilov — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

File: .github/workflows/ci-admin.yml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+on:
+  push:
+    branches:
+      - main
+      - release-*
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
+  pull_request:
+    paths-ignore:
+      - "agent/**"
```
Comment (BupycHuk) [inline .github/workflows/ci-admin.yml:12]: instead of `paths-ignore` I would use `paths`

**Pull request merged**